crates/client/sync/src/commitments/contracts.rs

```diff
@@ -30,50 +39,50 @@ pub struct ContractLeafParams {
 /// # Returns
 ///
 /// The storage root hash.
-pub fn update_storage_trie<B: BlockT>(
+pub fn update_storage_trie(
+    overrides: &OverrideHandle<Block<Header<u32, BlakeTwo256>, OpaqueExtrinsic>>,
     contract_address: &ContractAddress,
-    csd: &Arc<CommitmentStateDiff>,
-    bonsai_db: &Arc<BonsaiDb<B>>,
-) -> Result<Felt252Wrapper, BonsaiDbError> {
+    storage_updates: &IndexMap<StorageKey, StarkFelt>,
+    maybe_block_hash: Option<H256>,
+) -> Result<Felt252Wrapper, BonsaiStorageError<BonsaiDbError>> {
+    // Contract storage bonsai db is stored in-memory
     let config = BonsaiStorageConfig::default();
-    let bonsai_db = bonsai_db.as_ref();
-    let mut bonsai_storage: BonsaiStorage<BasicId, &BonsaiDb<B>, Pedersen> =
+    let bonsai_db = HashMapDb::<BasicId>::default();
+    let mut bonsai_storage =
         BonsaiStorage::<_, _, Pedersen>::new(bonsai_db, config).expect("Failed to create bonsai storage");
 
-    if let Some(updates) = csd.storage_updates.get(contract_address) {
-        for (storage_key, storage_value) in updates {
-            let key = BitVec::from_vec(Felt252Wrapper::from(storage_key.0.0).0.to_bytes_be()[..31].to_vec());
-            let value = Felt252Wrapper::from(*storage_value);
-            bonsai_storage.insert(key.as_bitslice(), &value.into()).expect("Failed to insert storage update into trie");
+    // Inserts old storage changes
+    let db = overrides.for_schema_version(&Undefined);
+    if let Some(hash) = maybe_block_hash {
+        if let Some(storage) = db.get_storage_from(hash, *contract_address) {
+            storage.into_iter().map(convert_storage).for_each(|(key, value)| {
+                if value != Felt252Wrapper::ZERO {
```

Comment:
will the case of a zero value be fixed in the bonsai crate? in this case add a todo

Reply:
Adding a TODO and sending the info over to the Bonsai Lib team :+1: